### PR TITLE
feat(players): Attackers

### PR DESCRIPTION
**Pull request by JDMJay**

### CODE REVIEW

File: src/views/players/player/player_attacker.cpp [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+			components::button("Send Buzzard", [] {
+				toxic::Spawn_Attackers(VEHICLE_BUZZARD, 100);
+			});
+			ImGui::SameLine();
+			components::button("Send Tank", [] {
+				toxic::Spawn_Attackers(VEHICLE_RHINO, 5);
+			});
+
+			components::button("Gooch Test", [] {
```
Comment (yasmasdas) [inline src/views/players/player/player_attacker.cpp]: "Gooch test?"
Rename to something like send gooch

File: src/views/players/player/player_attacker.cpp [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+				*script_global(1890140).at(244).at(1).as<Player*>() = g_player_service->get_selected()->id();
+				scripts::start_launcher_script(171);
+			});
+			ImGui::SameLine();
+			components::button("Mugger Test", [] {
```
Comment (yasmasdas) [inline src/views/players/player/player_attacker.cpp]: "Mugger test?"
Rename to something like send Mugger

File: src/util/toxic.hpp [hunk 3]
```diff
@@ -151,4 +153,39 @@ namespace big::toxic
...
+		PED::SET_PED_CONFIG_FLAG(ped, 281, true);
+
+		TASK::TASK_PLANE_MISSION(ped, veh, 0, ply, 0, 0, 0, 6, 0.0, 0.0, 0.0, 2500.0, -1500.0, 0);
+		TASK::TASK_VEHICLE_FOLLOW(ped, veh, ply, 540, 525117, 1);
+		TASK::TASK_VEHICLE_SHOOT_AT_PED(ped, ply, 100);
+
+		WEAPON::GIVE_WEAPON_TO_PED(ped, rage::joaat("WEAPON_RAILGUN"), 9999, true, true);
+		TASK::TASK_COMBAT_PED(ped, ply, 0, 16);
+
+		PED::SET_PED_FIRING_PATTERN(ped, 0xC6EE6B4C);
```
Comment (yasmasdas) [inline src/util/toxic.hpp]: rage::joaat("FIRING_PATTERN_FULL_AUTO") not 0xC6EE6B4C for readability

File: src/views/players/player/player_attacker.cpp [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+				*script_global(1853348).at(834).at(1).as<Player*>() = g_player_service->get_selected()->id();
```
Comment (yasmasdas) [inline src/views/players/player/player_attacker.cpp]: outdated gpbd_fm_1 global use [the class](https://github.com/Yimura/GTAV-Classes/blob/master/script/globals/GPBD_FM.hpp) like (code is example maybe not correct)
pid = ...
scr_globals::gpbd_fm_1.as<GPBD_FM*>()->Entries[self::id].GangCallRequestedServices = 0
scr_globals::gpbd_fm_1.as<GPBD_FM*>()->Entries[self::id].GangCallTarget = pid

File: src/views/players/player/player_attacker.cpp (hunk 1 above)
Comment (Yimura) [reply src/views/players/player/player_attacker.cpp]: I think this is an old test option that @maybegreat48 added.

Comment (JDMJay) [reply src/views/players/player/player_attacker.cpp]: yea I removed it from misc and added it into attackers

**Comment by binded-zz** [comment]
could you add the greifer jeus there was a few others that another menu had i for get what they was but i think chris something some Elvis  looking dude idk what there names was but they was cool 

some was just the dudes and some was with cars one was a bike and another was the truck with a ram on it 

we can already send attacker peds and body guards  
please can we have a few ready under players  i like to send them to dicks that are in god and killing every one every where's

wow after looking at what you added you added way more then i thought

File: src/views/players/player/player_attacker.cpp (hunk 4 above)
Comment (JDMJay) [reply src/views/players/player/player_attacker.cpp]: not sure if i have done this correctly so I'm going to leave it unresolved for now

Comment (yasmasdas) [reply src/views/players/player/player_attacker.cpp]: @JDMJay You did not do it correct. and to add to that what I gave was example code
so if my example works (untested) your code should be

```
components::button("Send Mugger", [] { 
scr_globals::gpbd_fm_1.as<GPBD_FM*>()->Entries[self::id].GangCallRequestedServices = 0;
scr_globals::gpbd_fm_1.as<GPBD_FM*>()->Entries[self::id].GangCallTarget = g_player_service->get_selected()->id();
});
```

**Comment by JDMJay** [comment]
> could you add the greifer jeus there was a few others that another menu had i for get what they was but i think chris something some Elvis looking dude idk what there names was but they was cool
> 
> some was just the dudes and some was with cars one was a bike and another was the truck with a ram on it
> 
> we can already send attacker peds and body guards please can we have a few ready under players i like to send them to dicks that are in god and killing every one every where's
> 
> wow after looking at what you added you added way more then i thought

I’m pretty sure tupoyemenu has that feature if @tupoy-ya  wants to contribute it he can I’m not gonna paste his code in for that sake but for now go use tupoyemenu if you want those type of trolling features etc

**Comment by binded-zz** [comment]
> > could you add the greifer jeus there was a few others that another menu had i for get what they was but i think chris something some Elvis looking dude idk what there names was but they was cool
> > some was just the dudes and some was with cars one was a bike and another was the truck with a ram on it
> > we can already send attacker peds and body guards please can we have a few ready under players i like to send them to dicks that are in god and killing every one every where's
> > wow after looking at what you added you added way more then i thought
> 
> I’m pretty sure tupoyemenu has that feature if @tupoy-ya wants to contribute it he can I’m not gonna paste his code in for that sake but for now go use tupoyemenu if you want those type of trolling features etc


i do use tupoyemenu he has the jeus one there but i think he said he broke it , i like his menu layout i can have the main menu and players menu open at the same times
-

there was a guy tyackman who had the other ones as well i want to say something like chris freeman or something like that 
his menu is now only ym base only now 

was just hoping i could see them in the main branch 
either way you added way more then i was even asking for if i read any of the stuff right  i think like send tanks, rc , lazer  
i dont remember 

but thanks for all of your hard work thank you

File: src/util/toxic.hpp [hunk 5]
```diff
@@ -151,4 +154,39 @@ namespace big::toxic
 		set_time_all((*g_pointers->m_network_time)->m_time + millis);
 	}
 
+	inline void Spawn_Attackers(Hash vehicle, float z)
```
Comment (Yimura) [inline src/util/toxic.hpp]: Function name doesn't follow code styling guidelines, please see CONTRIBUTING.md.

File: src/views/players/player/player_attacker.cpp [hunk 6]
```diff
@@ -0,0 +1,51 @@
+#include "core\scr_globals.hpp"
+#include "util/toxic.hpp"
+#include "views/view.hpp"
+
+#include <script\globals\GPBD_FM.hpp>
```
Comment (Yimura) [inline src/views/players/player/player_attacker.cpp]: Use forwards facing slashes for file paths.

**Review by Yimura — CHANGES_REQUESTED** [review]

File: src/views/players/player/player_attacker.cpp [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+#include "util/toxic.hpp"
+#include "views/view.hpp"
+
+#include <script/globals/GPBD_FM.hpp>
+
+namespace big
+{
+	void view::player_attacker()
+	{
+		if (ImGui::TreeNode("Attackers"))
```
Comment (maybegreat48) [inline src/views/players/player/player_attacker.cpp:11]: You should probably implement this in the ped spawner by adding an option to set the ped in a vehicle
The Gooch and the mugger can go in toxic

File: src/util/toxic.hpp [hunk 8]
```diff
@@ -151,4 +154,39 @@ namespace big::toxic
...
+		Vehicle veh = vehicle::spawn(vehicle, pos, heading, true);
+		VEHICLE::SET_VEHICLE_ENGINE_ON(veh, true, true, false);
+		VEHICLE::CONTROL_LANDING_GEAR(veh, 3);
+
+		static const Hash playerGroup = rage::joaat("PLAYER");
+		static const Hash civGroup    = rage::joaat("CIVMALE");
+		static const Hash femCivGroup = rage::joaat("CIVFEMALE");
+
+		Hash relationshipGroup;
+		PED::ADD_RELATIONSHIP_GROUP("_HOSTILE_JESUS", &relationshipGroup);
```
Comment (maybegreat48) [inline src/util/toxic.hpp:172]: Custom relgroups do not sync in MP
```c++
PED::SET_PED_RELATIONSHIP_GROUP_HASH(ped, RAGE_JOAAT("HATES_PLAYER"));
```
can be used instead

File: src/views/players/player/player_attacker.cpp (hunk 7 above)
Comment (JDMJay) [reply src/views/players/player/player_attacker.cpp:11]: This is just for presets I will do this is a different pr

File: src/util/toxic.hpp (hunk 8 above)
Comment (JDMJay) [reply src/util/toxic.hpp:172]: Kinda confused how to use this

Comment (maybegreat48) [reply src/util/toxic.hpp:172]: Replace
```c++
PED::SET_PED_RELATIONSHIP_GROUP_HASH(ped, relationshipGroup);
```
With
```c++
PED::SET_PED_RELATIONSHIP_GROUP_HASH(ped, RAGE_JOAAT("HATES_PLAYER"));
```
And remove
```c++
static const Hash playerGroup = rage::joaat("PLAYER");
static const Hash civGroup    = rage::joaat("CIVMALE");
static const Hash femCivGroup = rage::joaat("CIVFEMALE");

Hash relationshipGroup;
PED::ADD_RELATIONSHIP_GROUP("_HOSTILE_JESUS", &relationshipGroup);

PED::SET_RELATIONSHIP_BETWEEN_GROUPS(5, relationshipGroup, playerGroup);
PED::SET_RELATIONSHIP_BETWEEN_GROUPS(5, relationshipGroup, civGroup);
PED::SET_RELATIONSHIP_BETWEEN_GROUPS(5, relationshipGroup, femCivGroup);
```

File: src/views/players/player/player_attacker.cpp [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+				toxic::spawn_attackers(VEHICLE_RHINO, 5);
+			});
+
+			components::button("Send Gooch", [] {
+				*script_global(1890140).at(244).at(1).as<Player*>() = g_player_service->get_selected()->id();
+				scripts::start_launcher_script(171);
+			});
+			ImGui::SameLine();
+			components::button("Send Mugger", [] {
+				scr_globals::gpbd_fm_1.as<GPBD_FM*>()->Entries[self::id].GangCallRequestedServices;
```
Comment (yasmasdas) [inline src/views/players/player/player_attacker.cpp:43]: my bad for giving bad example that had a error last time but i did say it was untested.
But now what you did you just put the line and are accessing it but not doing anything with it.

you should use [Set(0) to set bit to 0](https://github.com/Yimura/GTAV-Classes/blob/master/script/types.hpp#L76) on GangCallRequestedServices as its type is a SCR_BITSET

**Comment by BeztNA** [comment]
anything new?

**Comment by Tayipc38** [comment]
How we spawn attackers feels quite limited to me. Id be up for a lil project to make it robust and customizable. Like vehicle choice, spawn manner, group size and maybe even track spawned attackers for extended features. A networked environment makes this quite a challenge, but I think I could manage something rather close to the scrips such as am_gang_call.

**Comment by yasmasdas** [comment]
> How we spawn attackers feels quite limited to me. Id be up for a lil project to make it robust and customizable. Like vehicle choice, spawn manner, group size and maybe even track spawned attackers for extended features. A networked environment makes this quite a challenge, but I think I could manage something rather close to the scrips such as am_gang_call.

I'm not a particular fan of this system either other then for the Send Mugger feature which still hasn't been correctly implemented.

**Comment by Rxann** [comment]
I do believe as though another attacker system(squad spawner) has been implemented, this can be closed.

**Comment by Yimura** [comment]
Stale PR

**Pull request closed without merging**